pubsub/natspubsub/nats.go

```diff
@@ -169,7 +169,11 @@ type TopicOptions struct{}
 
 // SubscriptionOptions sets options for constructing a *pubsub.Subscription
 // backed by NATS.
-type SubscriptionOptions struct{}
+type SubscriptionOptions struct {
```

Comment:
This allows you to set the Queue if you're using the regular constructor, or if you're using a custom URLOpener. How about adding some code near line 160 to support a "queue" URL parameter so that it can be done for the default URLOpener?

Something like this:

opts := o.SubscriptionOptions
for param, values := range u.Query() {
  if param == "queue" {
    opts.Queue = values[0]
  } else {
    return nil, fmt.Errorf("open subscription %v: invalid query parameter %s", u, param)
  }
}
...

Reply:
And add a test for this near line 381.

Reply:
Good thinking.

I also added a ```strings.ToLower(param)``` and a check for ```values != nil```, just in case anything goes wrong.

The URL to be decoded is ```"nats://mytopic?queue=queue1"``` and used on unit test ```TestOpenSubscriptionFromURL```

Also added an example ```Example_openQueueSubscriptionFromURL```